dataplane/README.md

```diff
@@ -0,0 +1,176 @@
+# The Lemming Dataplane
+
+The dataplane reads and writes packets from ports and the forwards packets between them.
+
+## Overview
+
+The dataplane consists of a several components:
+
+1. forwarding: packet processing actions, tables, and ports.
+1. saiserver: implements a "proto-ized" version of the [Switch Abstraction Interface API](https://github.com/opencomputeproject/SAI) and defines the forwarding pipeline
+1. apigen: code to generate the proto API for the saiserver and C++ code that calls the API
+1. dplanerc: gNMI reconciler that configures the dataplane based on OpenConfig schema
+1. standalone: shared library that a implements the C SAI API and container running only the dataplane
+
+There are two ways to run the dataplane. The first as part of Lemming, using the gNMI reconciliation. The second is a SAI implementation (e.g. in SONIC).
+
+## Components
+
+### forwarding
+
+The forwarding directory contains packet processing actions, tables, and ports. These are the building blocks of the dataplane: they can be combined to create the
+packet processing pipeline. The forwarding behavior is configured using the [gRPC API](../proto/forwarding/forwarding_service.proto).
+
+Generally, external users should not program the dataplane using this API directly. It is too flexible and low-table. Instead, the saiserver API should be used instead.
```

Comment:
low-table -> low-level?

Also mention here that when saiserver API is used, then don't use the forwarding API to avoid conflicts?